docs and code fixes for "Combinatorics" module

gxyd opened the pull request:
I will make a few changes in docs or code fixes as suitable(as i read more of the code of combinatorics module)

Commit:
[WIP] docs fixes

scopatz commented on sympy/combinatorics/tests/test_perm_groups.py:
```diff
@@ -691,3 +692,7 @@ def test_make_perm():
         Permutation([6, 7, 3, 2, 5, 4, 0, 1])
+
+
+def test_PermutationGroup():
```
PEP8: function names are `lowercase_with_underscores`

Commit:
Merge branch 'alternate_Master' into group_theory_PR1

Commit:
`.is_normal` for PermutationGroup with `strict`(optional)

Add `strict` argument for `.is_normal` method for the PermutationGroup,
similar to `strict` argument for `.is_subgroup` method, it also does the same
thing to make the degree of both the groups same, like
```
>>> s4 = SymmetricGroup(4)
>>> s6 = SymmetricGroup(6)
>>> s6.is_normal(s4, strict=False)
True
```

Earlier this used to raise IndexError, the default value of `strict` is True

Commit:
code quality fixes

gxyd commented:
@jksuom  please review.

jksuom commented on sympy/combinatorics/perm_groups.py:
```diff
@@ -123,10 +123,10 @@ def __new__(cls, *args, **kwargs):
         """
-        args = list(args[0] if is_sequence(args[0]) else args)
         if not args:
             raise ValueError('must supply one or more permutations '
```
It could be even possible to avoid ``ValueError``. One can think of the group generated by an empty args list as the symmetric group of the empty set, that consists of the identity element alone. So it might be possible to set
``args = [Permutation()]``
(assuming ``Permutation()`` is acceptable) and then to go on with no error message.

Commit:
PermutationGroup() with no argument contains only IdentityElm

Avoid the `ValueError` caused, since PermutationGroup() with no
arguments. One can think of the group generated by an empty
args list as the symmetric group of the empty set, that consists of the
identity element alone.

So set `args = [Permutation()]` and then to go on with no error message.
It should rather return `SymmetricGroup(0)` after PR #10363 is complete.

Commit:
`else` clause used

jksuom commented:
This looks good to me. Ready to merge when the tests pass.

jksuom commented:
Thanks. Merging.
